lec09-modeling-r.Rmd

```diff
@@ -1,4 +1,615 @@
+
+Think of the partial derivative as a slice in a particular direction through a potato: you could
+look at how the potato shape changes along its long direction or its short direction, and by doing 
+a partial derivative you are slicing along only one of the directions. 
+
+$$\frac{\partial f}{\partial y} = \frac{\partial }{\partial y} (ax-bxy) = - bx$$
+$$\frac{\partial g}{\partial x} = \frac{\partial }{\partial x} (cx-dy) = c$$
+
+$$\frac{\partial g}{\partial x} = \frac{\partial }{\partial x} (cx-dy) = -d$$
+
+We can put all these terms into the Jacobian. This matrix is now the Jacobian corresponding
+to our predator-prey model.
+
+$$J=\begin{bmatrix}
+    a - by & -bx \\
+    c & -d
+\end{bmatrix}$$
+
+### Classifying fixed points using eigenvalues of the Jacobian
+
+From here, we need to calculate the **eigenvalues** of the Jacobian matrix. The eigenvalues of a
+square matrix $A$ are numbers (often denoted $\lambda$) that satisfy the following relationship,
+where $\vec{v}$ is a vector whose length is the dimension of $A$. If $A$ is a 2-by-2 matrix, 
+$\vec{v}$ is a length-2 vector. 
+
+$$A\vec{v} = \lambda \vec{v}$$.
+
+What this means is that if the matrix $A$ is multiplied with a particular vector $\vec{v}$,
+the resulting vector is a number multiplied by the original vector. 
+
+This isn't true for just any vector. In general, this is how matrix multiplication works:
+
+$$\begin{bmatrix}
+    \alpha & \beta \\
+    \gamma & \delta
+\end{bmatrix} 
+\begin{bmatrix} 
+v_1 \\
+v_2
+\end{bmatrix} = 
+\begin{bmatrix}
+\alpha v_1 + \beta v_2 \\
+\gamma v_1 + \delta v_2
+\end{bmatrix}$$
+
+The result is a new vector, still of length 2, but not necessarily a constant multiplied by 
+the original vector $(v_1, v_2)$. 
+
+Our challenge is to find the eigenvalues of the Jacobian. We could also find the corresponding
+eigenvectors, but we don't need to for stability analysis.
+
+There is a formula to calculate eigenvalues: solve the equation $\text{Det}(A - \lambda \mathbb{1}) = 0$).
+$\text{Det}$ stands for 'determinant', and $\mathbb{1}$ is the **identity matrix**: a matrix of the same
+shape as $A$ but with $1$s on the diagonal and $0$s everywhere else.
+
+However, we won't calculate eigenvalues by hand. R has a function called `eigen` that we can use to
+directly calculate the eigenvalues of a matrix. 
+
+```{r}
+# stability analysis for predator-prey model
+
+# choose parameter values
+a <- 5
+b <- 1
+c <- 1
+d <- 0.2
+
+# start with a fixed point to look at
+y <- 0
+x <- 0
+
+# define the partial derivatives 
+dx_dx <- a - b*y
+dx_dy <- -b*x
+dy_dx <- c
+dy_dy <- -d
+
+# create the Jacobian matrix
+J<-matrix(c(dx_dx, dx_dy, dy_dx, dy_dy), nrow = 2, ncol = 2, byrow=TRUE);
+
+# calculate eigenvalues
+e_vals<-eigen(J)$values
+```
+
+There are two eigenvalues because our system is two-dimensional. What do they mean? For a set of eigenvalues ($\lambda_1$, $\lambda_2$, etc.):
```

Comment:
Is there a good explanation/image/animation that will help give intuition about this? *Why* do the eigenvalues infer these things about the fixed points?

Reply:
Yeah you've mentioned this a few times. I'm not sure how best to show it - in my opinion the clearest explanation of why this works is quite technical. I'll see what I can do!